src/dataflow/constructor.rs

```diff
@@ -273,6 +274,9 @@ pub(crate) fn node_manager_constructor<OP: Operator + 'static, B: Backend>(
                 .wait_timeout(std::time::Duration::from_millis(2000))
                 .expect("Failed to start NodeManager");
 
+            #[cfg(feature = "metrics")]
```

Comment:
Well this works too, but I meant moving the gauge! call to the NodeManager's ``on_start`` method. And described what code had to be moved in the constructor part for it to work properly.

Would preferably like to keep out metrics calls outside of dataflow construction.